notes/lecture17.tex

```diff
@@ -0,0 +1,189 @@
+\[
+    \limsup_{t \to \infty} -\|\nabla f(x_t)\|^2 < 0
+\]
+By \ref{eqn:4}, this must mean that $\eta_t \to 0$. This implies $\exists t_0$ such that $\forall t \ge t_0$
+\[
+    f(x_t) - f(x_t - \frac{\eta_t}{\gamma} \nabla f(x_t)) < \frac{\sigma \eta_t}{\gamma} \|\nabla f(x_t)\|^2
+\]
+Because $\eta_t \to 0$, we know that after some $t_0$ all step sizes are chosen with a $m\ge 1$. Therefore, going back one iteration of Armijo rule was not good enough to satisfy the inequality or else that or some previous step size would have been chosen. \\
+\\
+Now let $\tilde{\eta_t} = \frac{\eta_t}{\gamma}$ and we can continue as follows
+\begin{align}
+    \frac{f(x_t) - f(x_t - \tilde{\eta_t} \nabla f(x_t))}{\tilde{\eta_t}} &< \sigma \|\nabla f(x)\|^2 &\Rightarrow \nonumber \\
+    \nabla f(x_t - \tilde{\eta_t} \nabla f(x_t))^T \nabla f(x_t) &< \sigma \|\nabla f(x)\|^2 &\Rightarrow \label{eqn:5} \\
+    \|\nabla f(x_t) \|^2 &\le \sigma \|\nabla f(x_t) \|^2 & \label{eqn:6}
+\end{align}
+\\
+\ref{eqn:5} using Mean Value Theorem (MVT) \\
+\ref{eqn:6} by taking the limit as $\eta_t \to 0 \Rightarrow \tilde{\eta_t} \to 0$ \\
+\\
+This is a contradiction because $0 < \sigma < 1$. Therefore, the limit point $\bar{x}$ is a stationary point of $f$.
+\end{proof}
+Therefore, if we can use the Armijo rule to determine step sizes that guarantee that gradient descent will converge to a stationary point.
+\end{proof}
+
+\subsection{Saddle Points}
+
+Now that we have found that gradient descent will converge to a stationary point, how concerned should we be that the stationary point is not a local minimum?
+
+\begin{definition}[Saddle points]
+Saddle points are stationary points that are not local optimum.
+\end{definition}
+This means that if $f$ is twice differentiable then $\nabla^2 f(x)$ has both positive and negative eigenvalues.
+\subsubsection{How do saddle points arise?}
+In most non-convex problems there exists several local minima. This is clear to see in problems that have natural symmetry such as in a two layer fully connected neural networks. 
+\begin{center}
+\includegraphics[width=0.3\linewidth, height=0.3\linewidth]{figures/lecture_17_neural_network.png} 
+\end{center}
+Notice that any permutation of the units of the hidden layer would preserve the same function, so we have approximately $h!$ local minima. Typically a convex combination of two distinct local minimum in a non-convex problem is not a local minimum. In the case where $\nabla f(x)$ is differentiable, then by MVT we know that there must exist another stationary point between any two local minimum, so often there exists at least one saddle point between any two distinct local minimum. Hence, many local minimum tends to lead to many saddle points. \\
+\\
+However, recent work has demonstrated that saddle points are usually not a problem.
+\begin{enumerate}
+    \item The set of saddle points form a measure zero set for random initialization. \cite{DBLP:journals/corr/GeHJY15}
```

Comment:
The statement should be: Gradient descent does not converge to strict saddle points from a random initialization.